public/js/controllers/main.js

```diff
@@ -289,7 +303,10 @@
 
       // don't notify originator of new message
       if (c.username !== message.username) {
-        createNotification(message.username, message.message);
+        // check if user's setting still allows for Web Notifications
+        if (sessionStorage.getItem('sfm-notifications') === 'true') {
```

Comment:
Rather than getting this every time, save the value to a variable and check against that.